src/apps/relay/windows/Service.cpp

```diff
@@ -0,0 +1,255 @@
+/*!
+ * Service
+ * \author Kang Lin <kl222@126.com>
+ * \see https://learn.microsoft.com/windows/win32/services/service-program-tasks
+ */
+
+#include "Service.h"
+#include <AtlBase.h>
+#include <AtlConv.h>
+#include <fstream>
+#include <stdio.h>
+#include <tchar.h>
+#include <windows.h>
+
+struct WindowsService {
+  LPTSTR name; // Service name
+  fnServiceStart start;
+  fnServiceRun run;
+  fnServiceStop stop;
+
+  SERVICE_STATUS_HANDLE
+  handle; // Current service status handle. don't need close it. See
+          // https://learn.microsoft.com/windows/win32/api/winsvc/nf-winsvc-registerservicectrlhandlera?redirectedfrom=MSDN&devlangs=cpp&f1url=%3FappId%3DDev11IDEF1%26l%3DZH-CN%26k%3Dk(winsvc%252FRegisterServiceCtrlHandler)%3Bk(RegisterServiceCtrlHandler)%3Bk(DevLang-C%252B%252B)%3Bk(TargetOS-Windows)%26rd%3Dtrue
+  HANDLE hEvent;
+  SERVICE_STATUS status; // Current service status
+};
+WindowsService g_Service;
+
+void ServiceLog(char *msg) {
+  //*
+  USES_CONVERSION;
+  OutputDebugString(A2T(msg));
+  return; //*/
```

Comment:
Was this for debug or this is intended?

Reply:
And general question since I'm not a big expert on windows : I assume you tested it - could you add a few lines to the description on how you did it? Also, in separate PRs, add something to the documentation about how to use coturn as Windows service (how to install, how it is configured, etc). 

Thanks for a huge contribution!

Reply:
Yes. only for debug.

Reply:
how to use coturn as Windows service (how to install, how it is configured, etc). See: https://learn.microsoft.com/windows/win32/services/service-program-tasks

Reply:
The documentation is not for me but for users of coturn who want to install it as a service

Reply:
The documentation is for user and developer. more documentation see comments.